src/platform/plugins/shared/visualizations/public/utils/saved_visualize_utils.ts

```diff
@@ -195,7 +183,10 @@ export async function findListItems(
 
       if (config) {
         return {
-          ...config.toListItem(savedObject as any),
+          // TODO: understand why this SO can take any shape based on type?
+          // This conflicts with the type of `savedObject` value as `VisualizationSavedObject`.
+          // See test case titled 'uses type-specific toListItem function, if available'
+          ...config.toListItem(savedObject),
```

Comment:
Do we want to open an issue for that? or is something you want to investigate more in this PR?